front/lib/document_tracker.ts

```diff
@@ -0,0 +1,66 @@
+import { Op } from "sequelize";
+
+import { TrackedDocument } from "./models";
+
+export async function updateTrackedDocuments(
+  dataSourceId: number,
+  documentId: string,
+  documentContent: string
+) {
+  const emailPattern = "\\S+@\\S+\\.\\S+";
+
+  const dustTrackTagRegex = new RegExp(
+    "DUST_TRACK\\(\\s*((?:" + emailPattern + "\\s*,?\\s*)+)\\)",
+    "g"
+  );
+
+  const dustTrackTags = documentContent.match(dustTrackTagRegex);
+  if (!dustTrackTags) {
+    return;
+  }
+  const allEmails: Set<string> = new Set();
+  for (const dustTrackTag of dustTrackTags) {
+    // remove 'DUST_TRACK(' and ')' from the tag
+    const emailsInTag = dustTrackTag
+      .replace(/DUST_TRACK\(/, "")
+      .replace(/\)/, "");
+
+    // split emails by comma and map over them to remove any trailing or leading spaces
+    const emails = emailsInTag.split(",").map((email) => email.trim());
+    for (const email of emails) {
+      allEmails.add(email);
+    }
+  }
+
+  const emails = Array.from(allEmails);
+  const upsertTrackedDoc = async (email: string) => {
+    const exists = !!(await TrackedDocument.count({
+      where: {
+        dataSourceId,
+        documentId,
+        email,
+      },
+    }));
+    if (exists) {
+      return;
+    }
+    await TrackedDocument.create({
+      dataSourceId,
+      documentId,
+      email,
```

Comment:
We want to check that it matches an existing user (otherwise it's a data leak vector).

If the user does not exist, send an email saying that they have to register to Dust (no detail of workspace or anything to leak a minimal amount of info).